internal/ihop/os_release_layer_creator.go

```diff
@@ -0,0 +1,104 @@
+package ihop
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+// A OsReleaseLayerCreator can be used to construct a layer that includes /etc/os-release.
+type OsReleaseLayerCreator struct {
+	Def Definition
+}
+
+func (o OsReleaseLayerCreator) Create(image Image, _ DefinitionImage, _ SBOM) (Layer, error) {
+	img, err := image.ToDaemonImage()
+	if err != nil {
+		return Layer{}, err
+	}
+
+	tarBuffer, err := os.CreateTemp("", "")
+	if err != nil {
+		return Layer{}, err
+	}
+	defer tarBuffer.Close()
+	tw := tar.NewWriter(tarBuffer)
+
+	// find any existing /etc/ folder and copy the header
+	hdr, _, err := findFile(img, "etc/")
+	if err != nil {
+		return Layer{}, err
+	}
+	err = tw.WriteHeader(hdr)
+	if err != nil {
+		return Layer{}, err
+	}
+
+	// find any existing /etc/os-release file and copy the header
+	hdr, content, err := findFile(img, "etc/os-release")
+	if err != nil {
+		return Layer{}, err
+	}
+
+	buffer, err := updateOsRelease(content, o.Def)
+	if err != nil {
+		return Layer{}, err
+	}
+
+	err = tw.WriteHeader(&tar.Header{
+		Name: "etc/os-release",
+		Mode: hdr.Mode,
+		Size: int64(buffer.Len()),
+	})
+
+	if err != nil {
+		return Layer{}, err
+	}
+
+	_, err = io.Copy(tw, buffer)
+	if err != nil {
+		return Layer{}, err
+	}
+
+	err = tw.Close()
+	if err != nil {
+		return Layer{}, err
+	}
+
+	return tarToLayer(tarBuffer)
+}
+
+func updateOsRelease(content io.Reader, def Definition) (*bytes.Buffer, error) {
+	scanner := bufio.NewScanner(content)
+	updatedContent := map[string]string{}
+
+	for scanner.Scan() {
+		before, after, found := strings.Cut(scanner.Text(), "=")
+		if found {
+			updatedContent[before] = after
+		}
+	}
+	err := scanner.Err()
+	if err != nil {
+		return nil, err
+	}
+
+	updatedContent["PRETTY_NAME"] = fmt.Sprintf("%q", def.Name)
+	updatedContent["HOME_URL"] = fmt.Sprintf("%q", def.Homepage)
+	updatedContent["SUPPORT_URL"] = fmt.Sprintf("%q", def.SupportURL)
+	updatedContent["BUG_REPORT_URL"] = fmt.Sprintf("%q", def.BugReportURL)
```

Comment:
A couple thoughts on this section:
* Should we use `strconv.Quote`? https://pkg.go.dev/strconv#Quote
* If the values of the support and bug report URLs are empty, should we keep the original values?

Reply:
I think it's fine to put empty strings for `support` and `bug_url`, as the only way you would get empty values is explicitly putting empty values in the `stack.toml`, right? Assume that's the only path, it seems reasonable to support the stack author's intention of leaving those fields blank.